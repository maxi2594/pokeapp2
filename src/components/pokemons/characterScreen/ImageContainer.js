import React, { useState } from 'react'
import PropTypes from "prop-types";

export const ImageContainer = ({ data }) => {

    const [imageSrc, setImageSrc] = useState(data.sprites.front_default)


    const sprites = (Object.keys(data.sprites).map(key => ({
        sprite: [key],
        src: data.sprites[key]
    })))

    return (
        <div className='imageContainer'>
            <img
                style={{ width: 250 }}
                src={imageSrc}
                alt={data.pokemon.name}
            />
            <div className="imageSelector">
                <ul>
                    {
                        sprites.map(elem => (
                            elem.src ?
                                (<button
                                    key={elem.sprite}
                                    onClick={() => setImageSrc(elem.src)}
                                >{elem.sprite}</button>)
                                :
                                null
                        )
                        )
                    }
                </ul>
            </div>
        </div>
    )
}

ImageContainer.propTypes = {
    data: PropTypes.object.isRequired
}
