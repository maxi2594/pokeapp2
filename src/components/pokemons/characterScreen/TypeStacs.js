import React, { useEffect, useState } from 'react'
import { getPokes } from '../../../helpers/getPokesApi'

export const TypeStacs = ({ src }) => {

    const [data, setdata] = useState({
        data: null,
        error: null
    })

    // useEffect(() => {
    //     getPokes(src).then(data => data.damage_relations.double_damage_from.map(elem=> console.log(elem)))
    // }, [src]);

    return (
        data ?
            (<div className='typeStacs'>
                <div className='stats'>
                    <h3 style={{ width: '100%' }}>Damage Relations</h3>
                    <label> Doble Damage From: </label>
                    {/* {
                    data.damage_relations.double_damage_from.map( damage => (
                     <span>{damage.name}</span>       
                    ))
                } */}
                </div>
            </div>)
            :
            null
    )
}
