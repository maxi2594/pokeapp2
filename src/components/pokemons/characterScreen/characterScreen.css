.flexContainer {
  display: flex;
  flex-flow: wrap;
  width: 100%;
  height: 100%;
  background-color: transparent;
}
.imageContainer {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 40%;
  height: 100%;
  padding: 2rem;
}
.imageSelector {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
button {
  border-radius: 1rem;
  background-color: transparent;
  border: 2px solid grey;
  color: grey;
}
