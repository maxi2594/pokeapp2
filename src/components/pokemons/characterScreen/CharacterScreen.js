import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router'

import { getPokes } from '../../../helpers/getPokesApi'
import { ImageContainer } from './ImageContainer';
import { TypeStacs } from './TypeStacs';
import "./characterScreen.css";

export const CharacterScreen = () => {

    const [data, setData] = useState(null)
    const [typeStacs, settypeStacs] = useState({
        show: true,
        src: ''
    })

    const { id } = useParams()

    const handeSetTypeStac = (src) => {
        settypeStacs(s => ({
            ...s,
            src
        }))
    }

    useEffect(() => {

        const data = getPokes(`https://pokeapi.co/api/v2/pokemon-form/${id}`)
        data.then(data => {
            setData(data)
        })

    }, [id])

    return (
        data
        &&
        (<div className='flexContainer'>

            <ImageContainer data={data} />

            <div className='statsContainer'>
                <div className='stats'>
                    <label> name : </label>
                    <p data-testid="name">{data.name.toUpperCase()}</p>
                </div>

                <div className='stats'>
                    <label>Battle Only :</label>
                    <p> {data.is_battle_only ? 'SI' : 'NO'}</p>
                </div>

                <div className='stats'>
                    <label>Types:</label>
                    <div>
                        {
                            data.types.map(elem =>
                            (<button
                                key={elem.slot}
                                onClick={() => handeSetTypeStac(elem.type.url)}
                            >{elem.type.name}</button>)
                            )
                        }
                    </div>
                </div>
                {
                    typeStacs.show && (<TypeStacs src={data.types[0].type.url} />)
                }

            </div>
        </div>

        )
    )
}
