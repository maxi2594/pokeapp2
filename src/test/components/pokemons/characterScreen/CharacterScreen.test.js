import { render, screen, act, cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import { CharacterScreen } from "../../../../components/pokemons/characterScreen/CharacterScreen"
import { data } from "../../../fakeData/fakeData";

describe('test in <CharacterScreen/>', () => {



    beforeEach(() => {
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => ({ ...data })
            })
        )
        global.alert = jest.fn((e) =>
            console.log({ error: e })
        )
        String.replaceAll = jest.fn()
    })

    afterEach(cleanup);


    test('should show properly', async () => {

        const wrapper = shallow(
            <MemoryRouter>
                <CharacterScreen />
            </MemoryRouter>
        )

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').length).toEqual(0);
    })

    test('should have content after state update', async () => {
        await act(async () =>
            render(
                <MemoryRouter>
                    <CharacterScreen />
                </MemoryRouter>
            ))

        expect(screen.getByText(data.name.toUpperCase()))
            .toBeInTheDocument();
    })

})

