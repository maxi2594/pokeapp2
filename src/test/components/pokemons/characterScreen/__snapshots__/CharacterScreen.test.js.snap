// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <CharacterScreen/> should show properly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "y8ghtw",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "y8ghtw",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <CharacterScreen />
</Router>
`;
